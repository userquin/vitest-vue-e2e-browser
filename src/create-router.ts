import { createRouter as _createRouter, createWebHashHistory, createWebHistory } from 'vue-router'
import routes from 'virtual:generated-pages'

const defaultBaseUrl = import.meta.env.BASE_URL ?? '/'
const e2e = import.meta.env.MODE === 'test'

export function createRouter(baseUrl = defaultBaseUrl, testing = e2e) {
  return _createRouter({
    history: testing ? createWebHashHistory(baseUrl) : createWebHistory(baseUrl),
    routes,
  })
}
