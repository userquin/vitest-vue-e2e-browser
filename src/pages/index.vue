<script setup lang="ts">
// import HelloWorld from '../components/HelloWorld.vue'
</script>

<template>
  <div class="home">
    <h1>Home</h1>
    <p>Home page content goes here.</p>
    <RouterLink to="about">
      Go to About
    </RouterLink>
    <HelloWorld msg="e2e testing home" />
  </div>
</template>

<style>
.home {
  display: grid;
  place-items: center;
  grid-template-columns: 1fr;
}
</style>
