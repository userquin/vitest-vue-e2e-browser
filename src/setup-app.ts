import { createRouter, createWebHashHistory, createWebHistory } from 'vue-router'
import { createApp } from 'vue'
import './style.css'
import routes from 'virtual:generated-pages'
import App from './App.vue'

const defaultBaseUrl = import.meta.env.BASE_URL ?? '/'
const e2e = import.meta.env.MODE === 'test'

export function setupApp(baseUrl = defaultBaseUrl, testing = e2e) {
  console.log(testing)
  const router = createRouter({
    history: testing ? createWebHashHistory(baseUrl) : createWebHistory(baseUrl),
    routes,
  })

  return createApp(App).use(router)
}
