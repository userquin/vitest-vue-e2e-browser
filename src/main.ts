import { setupApp } from './setup-app'

setupApp().mount('#app')
