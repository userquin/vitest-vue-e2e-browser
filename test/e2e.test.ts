/// <reference types="vitest" />

import { nextTick } from 'vue'
import { setupApp } from '../src/setup-app'

describe('Testing App Routing', () => {
  let container: HTMLElement | undefined
  beforeAll(() => {
    const app = setupApp()
    container = document.createElement('div')
    container.id = 'app'
    document.body.appendChild(container)
    app.mount(container)
  })

  it('mounts', async () => {
    expect(container).toBeTruthy()
    expect(container.innerHTML).toContain('Home')
    // expect(container.innerHTML).toContain('e2e testing')
    expect(container.innerHTML).toMatchSnapshot()
    const link: HTMLAnchorElement = container.querySelector('a[href$="about"]')
    expect(link).toBeTruthy()
    link.click()
    await nextTick()
    expect(container.innerHTML).toContain('About')
  })
})
