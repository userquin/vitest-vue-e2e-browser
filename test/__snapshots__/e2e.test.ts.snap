// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing App Router > mounts 1`] = `"<div class=\\"home\\"><h1>Home</h1><p>Home page content goes here.</p><a href=\\"#/about\\" class=\\"\\"> Go to About </a></div>"`;

exports[`Testing App Routing > mounts 1`] = `"<div class=\\"home\\"><h1>Home</h1><p>Home page content goes here.</p><a href=\\"#/about\\" class=\\"\\"> Go to About </a></div>"`;
